### PR TITLE
Functions for on demand code execution added in Logger class

### PR DESCRIPTION
**Pull request by vagmcs**
* when[level] functions execute the given code block only if the respective
log level is enabled.

* return types added to some public functions. Named boolean flags added to some function calls in order to be more readable. :)

### CODE REVIEW

File: src/main/scala/com/typesafe/scalalogging/Logger.scala [hunk 1]
```diff
@@ -112,6 +112,8 @@ final class Logger private (val underlying: Underlying) extends Serializable {
 
   def error(marker: Marker, message: String, args: Any*): Unit = macro LoggerMacro.errorMessageArgsMarker
 
+  def whenError(body: Unit): Unit = macro LoggerMacro.errorCode
```
Comment (analytically) [inline src/main/scala/com/typesafe/scalalogging/Logger.scala]: would you be able to add this to the README?

File: src/main/scala/com/typesafe/scalalogging/Logger.scala [hunk 2]
```diff
@@ -93,7 +93,7 @@ object Logger {
 }
 
 /**
- * Implementation for a performant logger based on macros and an underlying `org.slf4j.Logger`.
+ * Implementation of a fast logger based on macros and an underlying `org.slf4j.Logger`.
```
Comment (analytically) [inline src/main/scala/com/typesafe/scalalogging/Logger.scala]: can you make this change to the README as well?

**Comment by analytically** [comment]
It's failing because of formatting.

**Comment by lightbend-cla-validator** [comment]
Hi @vagmcs,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: src/main/scala/com/typesafe/scalalogging/Logger.scala (hunk 1 above)
Comment (vagmcs) [reply src/main/scala/com/typesafe/scalalogging/Logger.scala]: I added an example for this.

File: src/main/scala/com/typesafe/scalalogging/Logger.scala (hunk 2 above)
Comment (vagmcs) [reply src/main/scala/com/typesafe/scalalogging/Logger.scala]: done

**Comment by vagmcs** [comment]
I signed the CLA.

File: src/main/scala/com/typesafe/scalalogging/Logger.scala [hunk 3]
```diff
@@ -126,6 +128,8 @@ final class Logger private (val underlying: Underlying) extends Serializable {
 
   def warn(marker: Marker, message: String, args: Any*): Unit = macro LoggerMacro.warnMessageArgsMarker
 
+  def whenWarn(body: Unit): Unit = macro LoggerMacro.warnCode
```
Comment (analytically) [inline src/main/scala/com/typesafe/scalalogging/Logger.scala]: let's name this whenWarnEnabled

Comment (vagmcs) [reply src/main/scala/com/typesafe/scalalogging/Logger.scala]: OK, I renamed all functions from when[level] to when[level]Enabled

**Comment by analytically** [comment]
Seems like Travis is still complaining about formatting, would you be able to fix that?

**Comment by vagmcs** [comment]
Now it seems ok.

**Comment by analytically** [comment]
Thanks for your contribution!

**Comment by cvaliente** [comment]
Cool, I was just looking for this feature. It was kind of confusing since the main readme already prominently features whenDebugEnabled but there's no actual published version of this library yet that includes this :)

**Comment by analytically** [comment]
Yeah sorry about that, will publish a release tonight.

**Pull request merged**